### ISSUE TITLE
Core: VF rotation functions and corresp. unit tests

### ISSUE THREAD

**Issue by GPMueller** [spirit-code/spirit#255]
The rotation function is currently wrong!
Unit tests are needed and in this context rotation functions for whole vectorfields could be created and tested:
```C++
void rotate(const scalar & angle, const Vector3 & axis, vectorfield & out);
void rotate(const scalar & angle, const vectorfield & axes, vectorfield & out);
```

**Comment by GPMueller** [comment]
This will be addressed with the Depondt Optimizer in PR #271.

**Comment by smavros** [comment]
Fixed with commit a2a3ccaa809c313771e6bd10374e75b1ba2878c8 in PR #271

**Comment by GPMueller** [comment]
Merged into `develop` with 1f8b20c7babc22760611f00467c04a95d36a0eb8.
